core/src/main/scala/org/apache/spark/sql/delta/commands/DeleteWithDeletionVectorsHelper.scala

```diff
@@ -293,8 +294,11 @@ object DeletionVectorBitmapGenerator {
       candidateFiles: Seq[AddFile],
       condition: Expression)
     : Seq[DeletionVectorResult] = {
+    val uriEncode = DeltaUDF.stringFromString(path => {
+      new Path(path).toUri.toString
+    })
     val matchedRowsDf = targetDf
-      .withColumn(FILE_NAME_COL, col(s"${METADATA_NAME}.${FILE_PATH}"))
+      .withColumn(FILE_NAME_COL, uriEncode(col(s"${METADATA_NAME}.${FILE_PATH}")))
```

Comment:
what is this change for?

Reply:
what does this fix? does it have test coverage?

Reply:
https://github.com/delta-io/delta/issues/1725

Tests fail without the fix. Adding a TODO in the code here to provide a more robust solution for 3.4.0 and 3.4.1